Vagrantfile

```diff
@@ -0,0 +1,68 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+INVENTORY_FILE = "inventory.vagrant"
+MEMORY_DEFAULT = 384
+HOSTS = [
+  { name: 'jenkins', ip: '172.18.10.9',
+    roles: %w{jenkins-master} },  
+]
+
+def generate_inventory(hosts, file)
+  roles = {}
+
+  File.open(file, "w") do |f|
+    hosts.each do |host|
+      host_string = "#{host[:name]} \
+internal_ip=#{host[:ip]} \
+external_ip=#{host[:ip]}.xip.io \
+ansible_ssh_host=#{host[:ip]}\n"
+
+      f.write(host_string)
+      host[:roles].each do |role|
+        roles[role] ||= []
+        roles[role] << host_string
+      end
+    end
+
+    roles.each do |role, hosts|
+      f.write("\n[#{role}]\n")
+      f.write(hosts.join)
+    end
+  end
+end
+
+generate_inventory(HOSTS, INVENTORY_FILE) if %w{up provision}.include?(ARGV[0])
```

Comment:
I don't think we need to do this if we have a single host and no host-level variables. I think we can use Vagrant's builtin `ansible.groups` instead: http://docs.vagrantup.com/v2/provisioning/ansible.html